services/src/datasets/create_from_workflow.rs

```diff
@@ -57,6 +57,7 @@ pub struct RasterDatasetFromWorkflowResult {
 impl TaskStatusInfo for RasterDatasetFromWorkflowResult {}
 
 pub struct RasterDatasetFromWorkflowTask<C: SessionContext> {
+    pub source_name: String,
```

Comment:
It is not possible to store the workflow id directly there, because the source can also be a layer id when calling "/layers/{provider}/{layer}/dataset" (layer_to_dataset).
I could create an enum that can either be a workflow id or a layer id and format it a string afterwards, but I think that would be overkill.